library/src/main/java/com/novoda/downloadmanager/PercentageBasedStorageRequirementRule.java

```diff
@@ -1,32 +1,31 @@
 package com.novoda.downloadmanager;
 
-import android.os.StatFs;
 import android.support.annotation.FloatRange;
 
 import java.io.File;
 
-public final class StorageRequirementsRule {
+public final class PercentageBasedStorageRequirementRule implements StorageRequirementRule {
 
     private final StorageCapacityReader storageCapacityReader;
     private final float percentageOfStorageRemaining;
 
-    public static StorageRequirementsRule withPercentageOfStorageRemaining(@FloatRange(from = 0.0, to = 0.5) float percentageOfStorageRemaining) {
-        return new StorageRequirementsRule(new StorageCapacityReader(), percentageOfStorageRemaining);
+    public static PercentageBasedStorageRequirementRule withPercentageOfStorageRemaining(@FloatRange(from = 0.0, to = 0.5) float percentageOfStorageRemaining) {
+        return new PercentageBasedStorageRequirementRule(new StorageCapacityReader(), percentageOfStorageRemaining);
     }
 
-    private StorageRequirementsRule(StorageCapacityReader storageCapacityReader,
-                                    @FloatRange(from = 0.0, to = 0.5) float percentageOfStorageRemaining) {
+    PercentageBasedStorageRequirementRule(StorageCapacityReader storageCapacityReader,
+                                          @FloatRange(from = 0.0, to = 0.5) float percentageOfStorageRemaining) {
         this.storageCapacityReader = storageCapacityReader;
         this.percentageOfStorageRemaining = percentageOfStorageRemaining;
     }
 
-    boolean hasViolatedRule(File storageDirectory,
-                            FileSize downloadFileSize) {
-        StatFs statFs = new StatFs(storageDirectory.getPath());
-        long storageCapacityInBytes = storageCapacityReader.storageCapacityInBytes(statFs);
+    @Override
+    public boolean hasViolatedRule(File storageDirectory,
+                                   FileSize downloadFileSize) {
+        long storageCapacityInBytes = storageCapacityReader.storageCapacityInBytes(storageDirectory.getPath());
         long minimumStorageRequiredInBytes = (long) (storageCapacityInBytes * percentageOfStorageRemaining);
         long usableStorageInBytes = storageDirectory.getUsableSpace();
-        long remainingStorageAfterDownloadInBytes = usableStorageInBytes - downloadFileSize.totalSize();
+        long remainingStorageAfterDownloadInBytes = usableStorageInBytes - downloadFileSize.remainingSize();
```

Comment:
It was using the `totalSize` to determine whether it could continue to download :man_facepalming: this is constant! It should be using the remaining size of the file :smile: